src/resolver.cpp

```diff
@@ -201,3 +239,41 @@ std::shared_ptr<ArWritableAsset> UsdOpenAssetIOResolver::_OpenAssetForWrite(
 
   return ArDefaultResolver::_OpenAssetForWrite(resolvedPath, writeMode);
 }
+
+std::optional<std::string> UsdOpenAssetIOResolver::locationForEntity(
```

Comment:
[fix] Return type is `ArResolvedPath`

Reply:
Whoops, left over from a previous incarnation. The return type of the method is `std::string`, but it's silently implicitly converted.

Reply:
Removed redundant `ArResolvedPath` wrapping in 44e5e35e

Reply:
Looking at it again, is this the right way around? Given that were using `value_or` with an `ArResolvePath`? See continuation of this in the other comment too.